tests/simple-server.rs

```diff
@@ -38,7 +39,8 @@ async fn hyper() {
 async fn upgrade_websocket(request: Request<Body>) -> Result<Response<Body>> {
 	assert!(hyper_tungstenite::is_upgrade_request(&request) == true);
 
-	let (response, stream) = hyper_tungstenite::upgrade(request, None)?;
+	let (response, stream) = hyper_tungstenite::upgrade(request, None)
+		.map_err(Error::Protocol)?;
```

Comment:
We might be able to change the error type here too. But it's only a test, so the error type isn't that important.

It's a shame tungstenite doesn't implement `Error: From<ProtocolError>` :(

Reply:
> It's a shame tungstenite doesn't implement `Error: From<ProtocolError>` :(

Oh right, I wanted to bring this up! This seems like the impl was just forgotten. One could probably just open a PR to add the impl.

Reply:
Indeed. I'm a little busy myself this week, so if you want to, knock yourself out :p Otherwise I may get round to such a PR eventually.

Reply:
Done: https://github.com/snapview/tungstenite-rs/pull/184

Reply:
Nice. Note that it's also missing for all other error variants, so it probably makes sense to add those too ;)